pam_authz/pam/README.md

```diff
@@ -0,0 +1,70 @@
+# Styra's PAM authorization module
+
+This implements a PAM module that integrates with Open Policy Agent (OPA)
+to make authorization decisions. When invoked, the module queries OPA
+and if no errors are returned, then the action is allowed. This allows
+administrators to implement fine-grained access control.
+
+This module is not intended to perform authentication. One can use standard
+linux accounts for that or a user database such as LDAP.
+
+
+## Usage
+
+This repo includes a detailed [example usage of the pam_authz module](../docker/README.md),
+so the instructions here focus on the installation and configuration of the
+PAM module. Details of writing policy to make the authorization decisions are
+found in the docker example.
+
+### Building the module
+
+To compile the authz PAM module, the following are needed:
+
+1. [golang compiler](https://golang.org/doc/install) (at least version 1.7)
+
+2. PAM development library. On debian, this can be obtained using
+   `apt-get install libpam0g-dev`.
+
+Finally, build the module using `make`. This will automatically obtain the go dependencies.
```

Comment:
Could consider moving the PAM module build into a container so avoid host setup issues.